core/services/workflows/syncer/handler.go

```diff
@@ -646,16 +587,24 @@ func (h *eventHandler) workflowActivatedEvent(
 // workflowDeletedEvent handles the WorkflowDeletedEvent event type. This method must remain idempotent.
 func (h *eventHandler) workflowDeletedEvent(
 	ctx context.Context,
-	payload WorkflowDeletedV1,
+	payload WorkflowDeletedEvent,
 ) error {
 	// The order in the handler is slightly different to the order in `tryEngineCleanup`.
 	// This is because the engine requires its corresponding DB record to be present to be successfully
 	// closed.
 	// At the same time, popping the engine should occur last to allow deletes to be retried if any of the
 	// prior steps fail.
-	key := EngineRegistryKey{Owner: payload.WorkflowOwner, Name: payload.WorkflowName}
+	key := EngineRegistryKey{
+		Owner:      payload.WorkflowOwner,
+		Name:       payload.WorkflowName,
+		WorkflowID: payload.WorkflowID.Hex(),
+		Version:    h.workflowRegistryVersion.Major(),
+	}
 
-	e, ok := h.engineRegistry.Get(key)
+	e, ok, err := h.engineRegistry.Get(key)
+	if err != nil {
+		return fmt.Errorf("engine registry error: %w", err)
```

Comment:
_nit_ not nice debug UX to repeat this suffix, maybe more specific on Get failure